ndb/MIGRATION_NOTES.md

```diff
@@ -123,6 +123,30 @@ The primary differences come from:
   is also a user-defined property named `key`. For an example, see the
   class docstring for `Model`.
 
+## Bare Metal
+
+One of the largest classes of differences comes from the use of the current
+Datastore API, rather than the legacy App Engine Datastore. In general, for
+users coding to the public interface, this won't be an issue, but users relying
+on pieces of the onstensibly private API that are exposed to the bare metal of
+the original datastore implementation will have to rewrite those pieces.
+Specifically, any function or method that dealt directly with protobuffers will
+no longer work. Datastore, itself, has moved to gRPC and protobuffers
+themselves are significantly different from legacy Datastore.  Additionally,
```

Comment:
I would rewrite this to just saw the `.protobuf` definitions have changed quite a bit from the barely-public API used by App Engine to the [public one][1] (also drop all usage of "protobuffers").

[1]: https://github.com/googleapis/googleapis/tree/master/google/datastore/v1

Reply:
ab2f965